url7f729337e3_timmolter_Sundial_tgz-pJ8-org_knowm_sundial_SampleRunJ8/src/org/quartz/core/JobRunShell.java

```diff
@@ -0,0 +1,372 @@
+ * </p>
+ *
+ * @see JobRunShellFactory
+ * @see org.quartz.core.QuartzSchedulerThread
+ * @see org.quartz.jobs.Job
+ * @see org.quartz.triggers.Trigger
+ * @author James House
+ */
+public class JobRunShell extends SchedulerListenerSupport implements Runnable {
+
+  /*
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Data members.
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   */
+
+  private JobExecutionContextImpl jec = null;
+
+  private QuartzScheduler qs = null;
+
+  private TriggerFiredBundle firedTriggerBundle = null;
+
+  private Scheduler scheduler = null;
+
+  private volatile boolean shutdownRequested = false;
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  /*
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Constructors.
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   */
+
+  /**
+   * <p>
+   * Create a JobRunShell instance with the given settings.
+   * </p>
+   *
+   * @param jobRunShellFactory A handle to the <code>JobRunShellFactory</code> that produced this <code>JobRunShell</code>.
+   * @param scheduler The <code>Scheduler</code> instance that should be made available within the <code>JobExecutionContext</code>.
+   * @param schdCtxt the <code>SchedulingContext</code> that should be used by the <code>JobRunShell</code> when making updates to the
+   *        <code>JobStore</code>.
+   */
+  JobRunShell(Scheduler scheduler, TriggerFiredBundle bndle) {
+
+    this.scheduler = scheduler;
+    this.firedTriggerBundle = bndle;
+  }
+
+  /*
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Interface.
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   */
+
+  @Override
+  public void schedulerShuttingdown() {
+
+    requestShutdown();
+  }
+
+  @Override
+  protected Logger getLog() {
+
+    return log;
+  }
+
+  void initialize(QuartzScheduler qs) throws SchedulerException {
+
+    this.qs = qs;
+
+    Job job = null;
+    JobDetail jobDetail = firedTriggerBundle.getJobDetail();
+
+    try {
+      job = qs.getJobFactory().newJob(firedTriggerBundle, scheduler);
+    } catch (SchedulerException se) {
+      qs.notifySchedulerListenersError("An error occured instantiating job to be executed. job= '" + jobDetail.getName() + "'", se);
+      throw se;
+    } catch (Throwable ncdfe) { // such as NoClassDefFoundError
+      SchedulerException se = new SchedulerException("Problem instantiating class '" + jobDetail.getJobClass().getName() + "' - ", ncdfe);
+      qs.notifySchedulerListenersError("An error occured instantiating job to be executed. job= '" + jobDetail.getName() + "'", se);
+      throw se;
+    }
+
+    this.jec = new JobExecutionContextImpl(scheduler, firedTriggerBundle, job);
+  }
+
+  private void requestShutdown() {
+
+    shutdownRequested = true;
+  }
+
+  @Override
+  public void run() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 72 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.